src/main/java/fi/otavanopisto/restfulptv/server/servicechannels/ServiceChannelEntityUpdater.java

```diff
@@ -110,25 +116,34 @@ public void onServiceIdUpdateRequest(@Observes ServiceChannelIdUpdateRequest eve
       }
     }
   }
-  
+
   @Timeout
   public void timeout(Timer timer) {
     if (!stopped) {
-      if (!queue.isEmpty()) {
-        String entityId = queue.iterator().next();
-        if (!queue.remove(entityId)) {
-          logger.warning(String.format("Could not remove %s from queue", entityId));
-        }
-        
-        Map<String, Object> serviceChannelData = serviceChannelResolver.loadServiceChannelData(entityId);
-        if (serviceChannelData != null) {
-          handleResponse(entityId, serviceChannelData);
-        } else {
-          logger.warning(String.format("Service channel %s caching failed", entityId));
+      if (running) {
+        return;
+      }
+
+      try {
+        running = true;
+        if (!queue.isEmpty()) {
+          String entityId = queue.iterator().next();
+          if (!queue.remove(entityId)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)